src/cast_core/src/SamplingLoader.cpp

```diff
@@ -138,15 +156,57 @@ void genny::actor::SamplingLoader::run() {
     }
 }
 
-SamplingLoader::SamplingLoader(genny::ActorContext& context)
+SamplingLoader::SamplingLoader(genny::ActorContext& context,
+                               std::string dbName,
+                               std::string collectionName,
+                               std::shared_ptr<DeferredSample> deferredSample)
```

Comment:
If you use `ShareableState` here, you don't need a custom Producer as well. Again, not sure of the implications.

Reply:
Oh of course now I am reading this after responding to your other comment :D I'll give that a go